app/invocation/invocation_cancel_button.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { Slash as SlashIcon } from "lucide-react";
+import Spinner from "../components/spinner/spinner";
+import { OutlinedButton } from "../components/button/button";
+import rpcService from "../service/rpc_service";
+import { invocation } from "../../proto/invocation_ts_proto";
+import errorService from "../errors/error_service";
+
+export interface InvocationCancelButtonComponentProps {
+  invocationId: string;
+}
+
+type State = {
+  isLoading?: boolean;
+};
+
+export default class InvocationCancelButtonComponent extends React.Component<InvocationCancelButtonComponentProps> {
+  state: State = {};
+
+  private onClick() {
+    this.setState({ isLoading: true });
+    rpcService.service
+      .cancelInvocation(new invocation.CancelInvocationRequest({ invocationId: this.props.invocationId }))
+      .catch((e) => errorService.handleError(e))
+      .finally(() => {
+        // Delay updating button loading state because after the Cancel RPC completes, can take some time for the
+        // invocation to be marked as disconnected
+        setTimeout(() => {
+          this.setState({ isLoading: false });
+        }, 2000);
```

Comment:
instead of this delay, you could have a state like "cancelled" and then set it to true when the RPC succeeds, and when in cancelled state the button can stay disabled so the user doesn't try to cancel again. So the `disabled` prop on the button would be like `disabled={this.state.isLoading || this.state.cancelled}`

You could also maybe add a `title` prop to the button which explains the reason that the button is still disabled even though it's no longer loading, e.g. `title={this.state.cancelled ? 'Invocation has already been cancelled and is now being cleaned up.' : undefined}`

The UI should automatically refresh once the invocation is disconnected and then the logic in the `InvocationButtons` component should kick in and render the re-run button instead.

Reply:
Done